From dbe95e198c936a144a419373f38e7c91fc192209 Mon Sep 17 00:00:00 2001
From: Jia Zhang <zhang.jia@linux.alibaba.com>
Date: Fri, 30 Apr 2021 03:50:28 +0000
Subject: [PATCH] Workaround the conflict between isgx out-of-tree and SGX
 in-tree Linux Driver

Linux kernel since 5.11 with SGX in-tree driver drops the support for the
SGX platforms without FLC hardware capability, e.g, SGX1 machine. However,
there are still non-trivial number of systems without FLC support. Please
read this link https://github.com/alibaba/inclavare-containers/blob/master/hack/no-sgx-flc/README.md
for more details.

If you are still the users of isgx out-of-tree Linux Driver, and you are
running a system with SGX in-tree driver, this is just for yours.

Signed-off-by: Jia Zhang <zhang.jia@linux.alibaba.com>
---
 sgx_main.c | 5 -----
 1 file changed, 5 deletions(-)

diff --git a/sgx_main.c b/sgx_main.c
index aa99564..518679c 100644
--- a/sgx_main.c
+++ b/sgx_main.c
@@ -298,11 +298,6 @@ static int sgx_drv_probe(struct platform_device *pdev)
 	if (boot_cpu_data.x86_vendor != X86_VENDOR_INTEL)
 		return -ENODEV;
 
-	if (!boot_cpu_has(X86_FEATURE_SGX)) {
-		pr_err("intel_sgx: the CPU is missing SGX\n");
-		return -ENODEV;
-	}
-
 	rdmsrl(MSR_IA32_FEAT_CTL, fc);
 
 	if (!(fc & FEAT_CTL_LOCKED)) {
-- 
1.8.3.1

